horace_core/Tobyfit/instpars_DGdisk.m

```diff
@@ -24,27 +24,59 @@
 % Get array of instruments
 instruments = header.instruments;
 nrun=instruments.n_runs;
-inst=repmat({instruments{1}},[nrun,1]);
-for i=2:nrun
-    inst{i}=instruments{i};;
-end
 
 % Fill output arguments
 ei=zeros(nrun,1);
 x0=zeros(nrun,1);
 xa=zeros(nrun,1);
 x1=zeros(nrun,1);
-mod_shape_mono=repmat(IX_mod_shape_mono,[nrun,1]);
-horiz_div=repmat(IX_divergence_profile,[nrun,1]);
-vert_div=repmat(IX_divergence_profile,[nrun,1]);
+
 for i=1:nrun
     ei(i)=header.expdata(i).efix;
-    x1(i)=abs(inst{i}.mono_chopper.distance);
-    x0(i)=abs(inst{i}.moderator.distance) - x1(i);          % distance from mono chopper to moderator face
-    xa(i)=abs(inst{i}.shaping_chopper.distance) - x1(i);    % distance from shaping chopper to mono chopper
-    mod_shape_mono(i)=inst{i}.mod_shape_mono;
-    horiz_div(i)=inst{i}.horiz_div;
-    vert_div(i)=inst{i}.vert_div;
 end
 
-end
+use_unique_objects = true;
+% switch between old and new inputs to object lookup
+% the old version will be removed after testing
```

Comment:
Is this tested?

Reply:
yes, by running the tobyfit tests in both modes